test/catch2/live_restore/api/test_live_restore_fs_open_file.cpp

```diff
@@ -0,0 +1,127 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * Tests for the Live Restore file system's fs_open_file function. [live_restore_open_file]
+ */
+
+#include "../utils_live_restore.h"
+
+using namespace utils;
+
+// Wrapper for fs_open_file. This will also check we get the expected return code.
+static WT_LIVE_RESTORE_FILE_HANDLE *
+open_file(live_restore_test_env &env, std::string file_name, WT_FS_OPEN_FILE_TYPE file_type,
+  int expect_ret = 0, int flags = 0)
+{
+    WT_SESSION *wt_session = reinterpret_cast<WT_SESSION *>(env.session);
+    WT_LIVE_RESTORE_FS *lr_fs = env.lr_fs;
+    WT_LIVE_RESTORE_FILE_HANDLE *lr_fh = nullptr;
+
+    int ret = lr_fs->iface.fs_open_file((WT_FILE_SYSTEM *)lr_fs, wt_session,
+      env.dest_file_path(file_name).c_str(), file_type, flags, (WT_FILE_HANDLE **)&lr_fh);
+    REQUIRE(ret == expect_ret);
+
+    return lr_fh;
+}
+
+TEST_CASE("Live Restore fs_open_file", "[live_restore],[live_restore_open_file]")
```

Comment:
I've only added simple tests here as I'm focused on unblocking WT-13792 - Support Logging with live restore. We can add tests for specific file flags and the other `WT_FS_OPEN_FILE_TYPE`s in the dedicated ticket from the test plan.